apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.ConformanceTests/CloudBigtableV2TestProxyImpl.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        string clientId = request.ClientId;
+        string projectId = request.ProjectId;
+        string instanceId = request.InstanceId;
+        string dataTarget = request.DataTarget;
+        CreateClientRequest.Types.SecurityOptions securityOptions = request.SecurityOptions;
+
+        CheckArgument(clientId is not ("" or null), "client id must be provided", context);
+        CheckArgument(projectId is not ("" or null), "project id must be provided", context);
+        CheckArgument(instanceId is not ("" or null), "instance id must be provided", context);
+        CheckArgument(dataTarget is not ("" or null), "data target must be provided", context);
+        CheckArgument(!securityOptions.UseSsl
+            || (securityOptions.SslRootCertsPem is not ("" or null)),
+            "security_options.ssl_root_certs_pem must be provided if security_options.use_ssl is true", context);
+
+        if (_idClientMap.ContainsKey(clientId))
+        {
+            context.Status = new Status(StatusCode.AlreadyExists, $"Client {clientId} already exists");
+            throw new RpcException(context.Status);
+        }
+
+        try
+        {
+            BigtableServiceApiSettings settings = new BigtableServiceApiSettings
+            {
+                AppProfileId = request.AppProfileId
+            };
+
+            if (request.PerOperationTimeout is not null)
+            {
+                Expiration exp = Expiration.FromTimeout(TimeSpan.FromSeconds(request.PerOperationTimeout.Seconds));
+                settings.CallSettings = CallSettings.FromExpiration(exp);
+            }
+
+            BigtableServiceApiClientBuilder builder = new BigtableServiceApiClientBuilder
+            {
+                Settings = settings
+            };
+            if (dataTarget != "emulator")
+            {
+                string accessToken = request.SecurityOptions.AccessToken;
+                builder.Endpoint = dataTarget;
+                builder.ChannelCredentials = GetChannelCredentials(securityOptions.UseSsl, securityOptions.SslRootCertsPem); ;
+                builder.GrpcChannelOptions = (securityOptions.UseSsl && securityOptions.SslEndpointOverride is not null)
+                    ? GrpcChannelOptions.Empty.WithCustomOption("grpc.ssl_target_name_override", securityOptions.SslEndpointOverride)
+                    : GrpcChannelOptions.Empty; ;
+                builder.GoogleCredential = accessToken is not null ? GoogleCredential.FromAccessToken(accessToken) : GoogleCredential.GetApplicationDefault();
+            }
+            else
+            {
+                builder.EmulatorDetection = EmulatorDetection.EmulatorOnly;
+            }
+            InstanceName instanceName = new InstanceName(projectId, instanceId);
+            GcpCallInvoker callInvoker = builder.CreateGcpCallInvoker();
+            builder.CallInvoker = callInvoker;
```

Comment:
Instead of doing this, after calling builder.BuildAsync you may have access to something called `LastCreatedChannel`.